linkml/generators/docgen.py

```diff
@@ -384,6 +384,10 @@ def uri(self, element: Element, expand=True) -> str:
         if isinstance(element, (EnumDefinition, SubsetDefinition)):
             # TODO: fix schema view to handle URIs for enums and subsets
             return self.name(element)
+
+        if self.hierarchical_class_view:
+            return self.schemaview.get_uri(element, expand=expand, use_element_type=True)
```

Comment:
Just a clarifying question – do we also need to check if `self.subfolder_type_separation` (`--subfolder-type-separation`) has been enabled?

Reply:
@sujaypatil96 Thanks for taking a look. I meant to use of course `self.subfolder_type_separation (--subfolder-type-separation)` not `self.hierarchical_class_view`. I did not recognize before as I always changed both at the same time. - Reviewing makes sense obviously  😄 - I will push a fix including a better test.